libutils/file_lib.c

```diff
@@ -993,9 +993,28 @@ FILE *safe_fopen_create_perms(
         return NULL;
     }
 
+    char fdopen_mode_str[3] = {0};
+    int fdopen_mode_idx = 0;
+
     int flags = 0;
     for (int c = 0; mode[c]; c++)
     {
+        /* Ignore mode letters not supported in fdopen() for the mode string
+         * that we will later use in fdopen(). Mode letters like 'x' will for
+         * example cause fdopen() to return error on Windows. According to the
+         * man page fdopen(3), the mode must be one of "r", "r+", "w", "w+",
+         * "a", "a+". Windows also have additional characters that are
+         * Microsoft extensions, but advice against using them to preserve
+         * ANSI portability. */
+        if (strchr("rwa+", mode[c]) != NULL)
+        {
+            if (fdopen_mode_idx >= sizeof(fdopen_mode_str))
```

Comment:
We need to make sure there's a `\0` at the end, right? So even `fdopen_mode_idx == 2` is a bad situation already.

Reply:
The buffer is initialized to all zeroes and we never write anything to the last byte, so it should be fine?

Reply:
Or maybe it should be `(fdopen_mode_idx >= (sizeof(fdopen_mode_str)) - 1)`